DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,189 @@
+/*
+ *
+* This is a part of CTPPS offline software.
+* Author:
+*   Fabrizio Ferro (ferro@ge.infn.it)
+*   Enrico Robutti (robutti@ge.infn.it)
+*   Fabio Ravera   (fabio.ravera@cern.ch)
+*
+*/
+
+#ifndef DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+#define DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelRecHit.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+
+
+#include "TVector3.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+//----------------------------------------------------------------------------------------------------
+
+class CTPPSPixelLocalTrack
+{
+  public:
+    class CTPPSPixelFittedRecHit: public CTPPSPixelRecHit
+    {
+      public:
+        CTPPSPixelFittedRecHit(const CTPPSPixelRecHit &hit, const TVector3 &space_point_on_det, std::pair<double,double> residual, std::pair<double,double> pull) :
+            CTPPSPixelRecHit(hit), space_point_on_det_(space_point_on_det), residual_(residual), pull_(pull), isUsedForFit_(false), isRealHit_(false) {}
+    
+        CTPPSPixelFittedRecHit() : CTPPSPixelRecHit(), residual_(std::pair<double,double>(0,0)), pull_(std::pair<double,double>(0,0)), isUsedForFit_(false), isRealHit_(false) {}
+    
+        virtual ~CTPPSPixelFittedRecHit() {}
+    
+        inline const TVector3 & getGlobalCoordinates() const { return space_point_on_det_; }
+        inline void setGlobalCoordinates(const TVector3 & space_point_on_det) { space_point_on_det_ = space_point_on_det; }
+    
+        inline double getXResidual() const { return residual_.first; }
+        inline double getYResidual() const { return residual_.second; }
+        inline void setXResidual(double residual) { residual_.first = residual; }
+        inline void setYResidual(double residual) { residual_.second = residual; }
+    
+        inline double getXPull() const { return pull_.first; }
+        inline double getYPull() const { return pull_.second; }
+        inline void setXPull(double pull) { pull_.first = pull; }
+        inline void setYPull(double pull) { pull_.second = pull; }
```

Comment:
set for pulls and coordinates appear to be redundant: the functionality is already in the constructors.

Reply:
I removed Set[X,Y]Residual and Set[X,Y] but I kept setGlobalCoordinates since it is used in RecoCTPPS/PixelLocal/src/RPixPlaneCombinatoryTracking.cc at line 316

Reply:
If I'm not mistaken, even there it can be avoided by calling an appropriate constructor at the right time (instead of a default one followed by a reset of parameters)